### PR TITLE
Ctda 1600

### PR DESCRIPTION
**Pull request by RobertLaverick**
Added GetBoxInfo endpoint

### CODE REVIEW

File: app/controllers/PushPullNotificationController.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      request.headers.get(Constants.XClientIdHeader) match {
+        case Some(clientId) => pushPullNotificationConnector.getBox(clientId).map {
+          response =>
+          response match {
+            case Left(error) if(error.statusCode == NOT_FOUND) => NotFound("No box found for your client id")
+            case Left(_) => InternalServerError("Unexpected Error")
+            case Right(box) => Ok(Json.toJson(box))
+          }
+        }
+        case None =>  Future.successful(BadRequest("Client Id Required"))
```
Comment (DavidGregory084) [inline app/controllers/PushPullNotificationController.scala:50]: We should use JSON errors in all the responses here

Comment (RobertLaverick) [reply app/controllers/PushPullNotificationController.scala:50]: So essentially just wrap it in a JsString like `BadRequest(JsString("Client..."))` ?

Comment (DavidGregory084) [reply app/controllers/PushPullNotificationController.scala:50]: No - the errors from our API need to follow the format from the Developer Hub [Reference Guide](https://developer.service.hmrc.gov.uk/api-documentation/docs/reference-guide#errors) - so they need to have a `code` and a `message` field. See e.g. https://github.com/hmrc/common-transit-convention-traders/blob/master/app/models/response/JsonErrorResponse.scala

Comment (DavidGregory084) [reply app/controllers/PushPullNotificationController.scala:50]: Or https://github.com/hmrc/common-transit-convention-guarantee-balance/blob/master/app/models/errors/BalanceRequestError.scala

**Review by DavidGregory084 — APPROVED** [review]

**Pull request merged**